pallets/relayer/src/lib.rs

```diff
@@ -0,0 +1,303 @@
+            }?;
+
+            Self::auth_accept_paying_key(signer.clone(), auth_by, user_key, paying_key, polyx_limit)
+        })
+    }
+
+    fn base_remove_paying_key(
+        origin: T::Origin,
+        user_key: T::AccountId,
+        paying_key: T::AccountId,
+    ) -> DispatchResult {
+        let PermissionedCallOriginData {
+            sender,
+            primary_did: sender_did,
+            ..
+        } = <Identity<T>>::ensure_origin_call_permissions(origin)?;
+
+        // allow: `origin == user_key` or `origin == paying_key`
+        if sender != user_key && sender != paying_key {
+            // allow: `origin == primary key of user_key's identity`
+            ensure!(
+                <Identity<T>>::get_identity(&user_key) == Some(sender_did),
+                Error::<T>::NotAuthorizedForUserKey
+            );
+        }
+
+        // Check if there is a paying key.
+        ensure!(
+            <Subsidies<T>>::contains_key(&user_key),
+            Error::<T>::NoPayingKey
+        );
+
+        // Get current Subsidy
+        let subsidy = <Subsidies<T>>::get(&user_key);
+
+        // Check if the current paying key matches.
+        ensure!(subsidy.paying_key == paying_key, Error::<T>::NotPayingKey);
+
+        // Remove paying key for user key.
+        <Subsidies<T>>::remove(&user_key);
+
+        Ok(())
+    }
+
+    fn base_update_polyx_limit(
+        origin: T::Origin,
+        user_key: T::AccountId,
+        polyx_limit: T::Balance,
+    ) -> DispatchResult {
+        let PermissionedCallOriginData {
+            sender: paying_key, ..
+        } = <Identity<T>>::ensure_origin_call_permissions(origin)?;
+
+        // Check if there is a paying key.
+        ensure!(
+            <Subsidies<T>>::contains_key(&user_key),
+            Error::<T>::NoPayingKey
+        );
+
+        // Get current Subsidy
+        let subsidy = <Subsidies<T>>::get(&user_key);
+
+        // Check if the current paying key matches.
+        ensure!(subsidy.paying_key == paying_key, Error::<T>::NotPayingKey);
+
+        // Update polyx limit
+        <Subsidies<T>>::mutate(&user_key, |subsidy| {
+            (*subsidy).remaining = polyx_limit;
+        });
+
+        Ok(())
+    }
+
+    /// Adds an authorization to add a `paying_key` to the `user_key`
+    pub fn unsafe_add_auth_for_paying_key(
+        from: IdentityId,
+        user_key: T::AccountId,
+        paying_key: T::AccountId,
+        polyx_limit: T::Balance,
+    ) -> u64 {
+        let auth_id = <Identity<T>>::add_auth(
+            from,
+            Signatory::Account(user_key.clone()),
+            AuthorizationData::AddRelayerPayingKey(
+                user_key.clone(),
+                paying_key.clone(),
+                polyx_limit,
+            ),
+            None,
+        );
+        Self::deposit_event(RawEvent::PayingKeyAuthorized(
+            from.for_event(),
+            user_key,
+            paying_key,
+            auth_id,
```

Comment:
If we keep this extrinsic, then let's include the `polyx_limit` too in the event.